### PR TITLE
mixerplugin: move up the list

### PR DESCRIPTION
**Pull request by kyessenov**
If we want telemetry to observe effects of the other pilot plugins, then mixer plugin needs to be first. There is a dependency on authn plugin, so it is right after it.

/assign @douglas-reid

### CODE REVIEW

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -107,9 +107,9 @@ var (
 	DefaultPlugins = []string{
 		plugin.Authn,
 		plugin.Authz,
+		plugin.Mixer,
 		plugin.Envoyfilter,
 		plugin.Health,
```
Comment (douglas-reid) [inline pilot/pkg/bootstrap/server.go]: I *feel* like having this happen after `Health` was intentional. This prevents policy/monitoring on health checks (and prometheus scrapes), I believe.

Do I have that right, @objectiser  ?

Comment (rshriram) [reply pilot/pkg/bootstrap/server.go]: Move health to top. Its to prevent tracing for health check calls.

Comment (kyessenov) [reply pilot/pkg/bootstrap/server.go]: Can you summarize what health does? Does it just put some header? Does it block requests?

Comment (objectiser) [reply pilot/pkg/bootstrap/server.go]: Yes the health check filter disables tracing - so needs to occur before the mixer.

@kyessenov  The health check filter just disables sampling of the active span, which is then propagated to any child spans within the proxy or called services.

Comment (kyessenov) [reply pilot/pkg/bootstrap/server.go]: Ok, moved health up prior to mixer.
